docs/contribution/releases/building-and-publishing.md

```diff
@@ -0,0 +1,48 @@
+   * Keep in mind that this should not be the version you’re building but the one before, as the stable tag will be updated later in the process.
+
+## Building WooCommerce
+
+1. Run the [“Release: Compile changelog” workflow”](https://github.com/woocommerce/woocommerce/actions/workflows/release-compile-changelog.yml) to generate the changelog.
+   1. Run from the workflow from trunk and enter the major version number.
+   2. Once the workflow finishes, it should’ve created 2 new PRs:
+      * One against trunk removing processed entries.
+      * One against the release branch, both removing processed entries and updating the readme.txt with the updated changelog.
+   3. Review and merge the PRs, updating if necessary.
+      * Ensure that the date in the changelog is also correct.
+2. (Skip for `-rc.1`) Run the version-bump workflow to update the WooCommerce version in relevant files on the release branch ahead of the release:
+   * This command will give you a link to a PR. Review and merge the PR once CI passes.
+3. Build the release ZIP file by running the [“Release: Build ZIP file” workflow](https://github.com/woocommerce/woocommerce/actions/workflows/release-build-zip-file.yml).
+   1. Run the workflow from the release branch and set "Create a draft GitHub release" to `true`.
+   2. Once the workflow is complete, the run will create a [draft release tag](https://github.com/woocommerce/woocommerce/releases) with an attached woocommerce.zip file.
+
+## Publishing the Release
+
+1. Run the ["Release: Upload release to WordPress.org" workflow](https://github.com/woocommerce/woocommerce/actions/workflows/release-upload-to-wporg.yml) to publish the release on WordPress.org
+   * Run the workflow from `trunk` and enter the tag of the release that was created above, e.g. 10.0.1.
+   * This step will create a new tag in SVN with the new release version and, if this release is a newer release than the current version in trunk, it will overwrite trunk. For example, if the latest trunk version is 10.1.0 and release 10.0.4 is published, it will not overwrite trunk. 
+2. When the workflow is complete, visit [https://wordpress.org/plugins/developers/releases/](https://wordpress.org/plugins/developers/releases/) and approve the release. It will take a few minutes for WordPress.org to build teh new version.
+3. Validate that the new release is available:
+   * https://plugins.svn.wordpress.org/woocommerce/tags/ should list the version number.
+   * The new version should also be available inside the Previous versions dropdown [inside the Advanced Options screen on the WordPress.org plugin page](https://wordpress.org/plugins/woocommerce/advanced/).
+4. If this is not an `RC` release, [Deploy the release to our staging environments](https://wp.me/PCYsg-18BQ) and monitor for any issues.
```

Comment:
Should we add a step to bump the stable tag using [this workflow](https://github.com/woocommerce/woocommerce/actions/workflows/release-update-stable-tag.yml) as well?